### PR TITLE
[LA.UM.9.12.r1] treewide: Linux 4.19.148

### PR DESCRIPTION
**Pull request by konradybcio**

### CODE REVIEW

**Review by MarijnS95 — CHANGES_REQUESTED** [review]
Just like the usual mainline review:

image


Clarify _why_ changes are made. That way everyone can follow along and confirm eventual observations/claims.

**Comment by konradybcio** [comment]
After some more showing git who's the boss here it turned out that this revert was in fact not necessary.

**Comment by MarijnS95** [comment]
Sounds good, but why is it not squashed in with the relevant patch release where it landed?

**Comment by MarijnS95** [comment]
In other words, how can something change from a revert to an "apply"?

**Comment by konradybcio** [comment]
I skipped it when applying .140 changes and at this point I am kind of scared to mess with git more..

**Comment by konradybcio** [comment]
I skipped it when applying .140 changes and at this point I am kind of scared to mess with git more..

**Comment by MarijnS95** [comment]
```console
konrad@msm4.19/kernel $ git rebase -i sony/aosp/LA.UM.9.12.r1  
```
```
pick f7c4bb5e394c treewide: Linux 4.19.139
pick 87805ce7ade4 treewide: Linux 4.19.140
f ed1c4a140010 net: refactor bind_bucket fastreuse into helper
pick 325888a897e2 treewide: Linux 4.19.141
pick 978424b4add3 treewide: Linux 4.19.142
pick d40b9516fb44 treewide: Linux 4.19.143
pick 61b8e140d49d treewide: Linux 4.19.144
pick 164b54ed1722 treewide: Linux 4.19.145
pick 19b68dfd15ec treewide: Linux 4.19.146
pick 7ca5ca93d192 treewide: Linux 4.19.147
pick 32cd91230522 treewide: Linux 4.19.148
```

Nothing scary about this.

File: drivers/base/power/main.c [hunk 1]
```diff
@@ -1760,13 +1760,17 @@ static int __device_suspend(struct device *dev, pm_message_t state, bool async)
...
+	 * to complete and if there's a runtime resume request pending for it,
+	 * resume it before proceeding with invoking the system-wide suspend
+	 * callbacks for it.
+	 *
+	 * If the system-wide suspend callbacks below change the configuration
+	 * of the device, they must disable runtime PM for it or otherwise
+	 * ensure that its runtime-resume callbacks will not be confused by that
+	 * change in case they are invoked going forward.
 	 */
-	if (pm_runtime_barrier(dev) && device_may_wakeup(dev))
```
Comment (kholk) [inline drivers/base/power/main.c:1768]: This is scary and must be tested very .. very carefully

File: drivers/pci/probe.c [hunk 2]
```diff
@@ -348,6 +348,57 @@ static void pci_read_bases(struct pci_dev *dev, unsigned int howmany, int rom)
 	}
 }
 
+static void pci_read_bridge_windows(struct pci_dev *bridge)
```
Comment (kholk) [inline drivers/pci/probe.c:351]: Check the modem on EDO, if Qualcomm had fantasy about the hypervisor restrictions, this may KO the platform...

**Pull request merged**